tpu-client/src/tpu_client.rs

```diff
@@ -95,6 +105,66 @@ where
         self.invoke(self.tpu_client.try_send_transaction(transaction))
     }
 
+    /// Serialize and send transaction to the current and upcoming leader TPUs according to fanout
+    /// size
+    /// Attempts to send and confirm tx "tries" times
+    /// Waits for signature confirmation before returning
+    /// Returns the transaction signature
+    pub fn send_and_confirm_transaction_with_retries<T: Signers + ?Sized>(
+        &self,
+        keypairs: &T,
+        transaction: &mut Transaction,
+        tries: usize,
+        pending_confirmations: usize,
+    ) -> TransportResult<Signature> {
+        for x in 0..tries {
+            let now = Instant::now();
+            let mut num_confirmed = 0;
+            let mut wait_time = MAX_PROCESSING_AGE;
+            // resend the same transaction until the transaction has no chance of succeeding
+            let wire_transaction =
+                bincode::serialize(&transaction).expect("transaction serialization failed");
+
+            while now.elapsed().as_secs() < wait_time as u64 {
+                let leaders = self
```

Comment:
I don't fully understand why you want to use leaders and connection cache directly instead of using `TpuClient::send_wire_transaction`

Reply:
good question. this is part of the issue I don't fully understand. If `send_write_transaction()` uses the non blocking TPUClient. Down the stack it calls: 
```
let conn = connection_cache.get_nonblocking_connection(addr);
conn.send_data(&wire_transaction).await
``` 
And this gives the error described above in the PR description. 
So i changed tpu_client to call: `cache.get_connection(tpu_address)` which calls `connection.new_blocking_connection()`. And this blocking connection works. but the nonblocking does not

Reply:
Do you know if this new method is going to be used only for integration tests?

Reply:
it is concerning for sure. Although the concern I have is not necessarily with `send_and_confirm_transaction_with_retries()` but with `try_send_transaction()`, `send_wire_transaction()`, and the nonblocking tpu client. Since calling `send_wire_transaction()` even on its own or with a couple retries fails.

Reply:
If this function is supposed to be used for integration tests, I would add some comments explaining the situation and proceed.

Reply:
ya its just used/been tested with integration tests. I just added a note for this function explaining.